### DISCUSSION TITLE
stumpy.motifs IndexError and unexpected results

### DISCUSSION THREAD

**Discussion by chvalovak** [TDAmeritrade/stumpy#784] [Q&A] [answered]
Hello, 
I am trying to use `stupmy.motifs` but I encountered a problem. With this setting of `stumpy.motifs` 

`    dists, indices = stumpy.motifs(timeseries["value"], mp['mp1'], max_matches=5, max_motifs=20)`

it produces 5 motifs with 5 matches and then this error occures:

`IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed`
![stumpy_err_pic](https://user-images.githubusercontent.com/46556207/213716984-ed5ff67e-2862-44c9-a968-2bbf8f29ecf8.png)

The array query_matches is empty (error occures in motifs.py).

I tried running `stumpy.motifs`  with benevolent parameters, like this:

`dists, indices = stumpy.motifs(timeseries["value"], mp['mp1'],max_matches=5, max_motifs=20, cutoff=np.inf, max_distance=np.inf, min_neighbors=1)`

This setting did not invoke the error but it discovered only ONE motif. Any idea what can cause this behavior?

Thanks in advance for any reply!

**Comment by seanlaw** [comment, marked as answer]
@chvalovak Thank you for your question and welcome to the STUMPY community. Are you able to provide a small sample time series and matrix profile input that reproduces the error? Without your inputs, it is hard to know what is causing the error

**Reply by NimaSarajpoor** [reply]
@seanlaw 
As you said, it is hard to say without having the inputs. So, what I am about to say is just a guess.


I think I encountered a similar issue before. And, we fixed it. See below:

https://github.com/TDAmeritrade/stumpy/blob/79593508346b20847a1e9b3e6e3fc91fdddcce5a/stumpy/motifs.py#L126-L134

`Lines 130-131` were added to prevent this issue. But, this if-block is not shown in the image provided by @chvalovak.

**Reply by NimaSarajpoor** [reply]
@seanlaw 
As a follow up to my previous comment, I would like to add something here:

IF `len(query_matches)==0`, then the for-loop in lines 133-134 has only one iteration, where `idx` is `candidate_idx`. Therefore, according to the following lines of code, `candidate_idx` will be updated to have the value `np.argmin(P[-1])`.   

https://github.com/TDAmeritrade/stumpy/blob/79593508346b20847a1e9b3e6e3fc91fdddcce5a/stumpy/motifs.py#L133-L136

Is that meaningful in such case? We may want to investigate this.

**Reply by chvalovak** [reply]
Thank you very much for your reply. Adding lines

```
if len(query_matches) == 0:  # pragma: no cover 
     query_matches = np.array([[np.nan, candidate_idx]]) 
```

helped. Motifs matching run without error.
